compiler/src/main/scala/org/scalajs/core/compiler/GenJSExports.scala

```diff
@@ -627,6 +618,29 @@ trait GenJSExports extends SubComponent { self: GenJSCode =>
       }
     }
 
+    private def reportCannotDisambiguateError(alts: List[Exported]): Unit = {
+      val currentClass = currentClassSym.get
+
+      // Find a position that is in the current class for decent error reporting
+      val pos = alts.collectFirst {
+        case ExportedSymbol(sym) if sym.owner == currentClass => sym.pos
+        case alt: ExportedBody                                => alt.pos
+      }.getOrElse {
+        currentClass.pos
+      }
```

Comment:
See the comment above the `collectFirst`? We're trying to find one of the alternatives that is actually defined in the current class to get its position. If there is no such alternative (because they are all inherited), we need to fall back on the class' position, otherwise we report an error potentially in a different file, which might not even be part of the current compilation run (bad).